src/platform/ESP32/nimble/BLEManagerImpl.h

```diff
@@ -0,0 +1,236 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Provides an implementation of the BLEManager singleton object
+ *          for the ESP32 platform (NimBLE).
+ */
+
+#ifndef BLE_MANAGER_IMPL_H
+#define BLE_MANAGER_IMPL_H
+
```

Comment:
I'm not a huge fan of #ifdef in implementation files as it makes code hard to follow. But since this file is just a header and since the only differences with src/platform/ESP32/BLEManagerImpl.h are #includes method calls related to BLE events should we use #ifdef instead of adding a new file ?